apps/web/src/components/ExperienceWorkStreams/ExperienceWorkStreams.tsx

```diff
@@ -0,0 +1,295 @@
+import { useIntl } from "react-intl";
+import PlusCircleIcon from "@heroicons/react/24/solid/PlusCircleIcon";
+import PencilSquareIcon from "@heroicons/react/20/solid/PencilSquareIcon";
+import TrashIcon from "@heroicons/react/20/solid/TrashIcon";
+import { useFormContext } from "react-hook-form";
+
+import { Button, Heading, Well } from "@gc-digital-talent/ui";
+import { unpackMaybes } from "@gc-digital-talent/helpers";
+import { FragmentType, getFragment, graphql } from "@gc-digital-talent/graphql";
+import { commonMessages } from "@gc-digital-talent/i18n";
+
+import pageTitles from "~/messages/pageTitles";
+import BoolCheckIcon from "~/components/BoolCheckIcon/BoolCheckIcon";
+import { WorkFormValues } from "~/types/experience";
+
+import ExperienceWorkStreamsEditDialog from "./ExperienceWorkStreamsEditDialog";
+import { WorkStreamWithoutKey } from "./types";
+
+const ExperienceWorkStreamsCommunity_Fragment = graphql(/* GraphQL */ `
+  fragment ExperienceWorkStreamsCommunity on Community {
+    id
+    name {
+      localized
+    }
+    workStreams {
+      id
+      name {
+        localized
+      }
+    }
+  }
+`);
+
+interface ExperienceWorkStreamsProps {
+  communitiesQuery?: FragmentType<
+    typeof ExperienceWorkStreamsCommunity_Fragment
+  >[];
+}
+
+const ExperienceWorkStreams = ({
+  communitiesQuery,
+}: ExperienceWorkStreamsProps) => {
+  const intl = useIntl();
+
+  const { setValue, watch } = useFormContext<WorkFormValues>();
+
+  const watchWorkStreams = watch("workStreams") ?? [];
+
+  const communities = getFragment(
+    ExperienceWorkStreamsCommunity_Fragment,
+    communitiesQuery,
+  );
+
+  const communitiesWithWorkStreams =
+    communities?.filter((item) => unpackMaybes(item?.workStreams).length > 0) ??
+    [];
+
+  const workStreamsByCommunity = new Map<string, WorkStreamWithoutKey[]>();
+  watchWorkStreams.forEach((item) => {
+    let workStream;
+    const community = communities?.find((c) => {
+      const matchedWorkStream = c.workStreams?.find((s) => s.id === item);
+      workStream = matchedWorkStream;
+      return !!matchedWorkStream;
+    });
+    if (community && workStream) {
+      const newWorkStreams = [
+        ...(workStreamsByCommunity.get(community.id) ?? []),
+        workStream,
+      ];
+      workStreamsByCommunity.set(community.id, newWorkStreams);
+    }
+  });
+
+  const handleUpdate = (workStreamIds: string[]) => {
+    setValue("workStreams", workStreamIds, { shouldDirty: true });
+  };
+
+  const handleRemove = (communityId?: string) => {
```

Comment:
This guy is broken now. It doesn't invoke the api and removes the community only on the front end. Could you check?

Reply:
oh...changes are not immediate now .. it waits until we hit save button. It does work after the save. I guess this is fine. Avoids a network delay.

Reply:
> oh...changes are not immediate now .. it waits until we hit save button. It does work after the save. I guess this is fine. Avoids a network delay.

We have to do it this way because when using the create an experience form, there is no experience to save the work streams selected to until the experience itself is created in the database.